File: consultoria/src/main/java/br/com/sigo/consultoria/services/EmpresaService.java
package br.com.sigo.consultoria.services;

import br.com.sigo.consultoria.domain.Empresa;
import br.com.sigo.consultoria.dtos.EmpresaDTO;
import br.com.sigo.consultoria.exceptions.ConsultoriaException;

import java.util.Optional;

public interface EmpresaService {

  Optional<Empresa> encontraEmpresa(String cnpj) throws ConsultoriaException;

  EmpresaDTO atualizar(EmpresaDTO empresa) throws ConsultoriaException;

  boolean atribuirCategoriaAEmpresa(String cnpj, Integer categoria) throws ConsultoriaException;
}

File: consultoria-app/app/src/main/java/br/com/sigo/consultoria/services/EmpresaService.kt
package br.com.sigo.consultoria.services

import br.com.sigo.consultoria.dtos.EmpresaDTO
import br.com.sigo.consultoria.internet.Response
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.HeaderMap
import retrofit2.http.POST

interface EmpresaService {

    @POST("api/empresa")
    fun atualizaEmpresa(
        @Body empresaDTO: EmpresaDTO, @HeaderMap headers: Map<String, String>
    ): Call<Response<EmpresaDTO>>

}